cmd/clusterctl/examples/vsphere/generate-yaml.sh

```diff
@@ -98,34 +98,88 @@ mkdir -p ${OUTPUT_DIR}
 
 # all variables used for yaml generation
 
-CLUSTER_NAME=${CLUSTER_NAME:-vsphere-cluster}
-SERVICE_CIDR=${SERVICE_CIDR:-100.64.0.0/13}
-CLUSTER_CIDR=${CLUSTER_CIDR:-100.96.0.0/11}
-
-VSPHERE_USER=${VSPHERE_USER:-}
-VSPHERE_PASSWORD=${VSPHERE_PASSWORD:-}
-VSPHERE_SERVER=${VSPHERE_SERVER:-}
-VSPHERE_MANAGER_IMG=${VSPHERE_MANAGER_IMG:-}
+export CLUSTER_NAME=${CLUSTER_NAME:-capv-mgmt-example}
+export SERVICE_CIDR=${SERVICE_CIDR:-100.64.0.0/13}
+export CLUSTER_CIDR=${CLUSTER_CIDR:-100.96.0.0/11}
+
+export VSPHERE_USER=${VSPHERE_USER:-}
+export VSPHERE_PASSWORD=${VSPHERE_PASSWORD:-}
+export VSPHERE_SERVER=${VSPHERE_SERVER:-}
+export CAPV_MANAGER_IMAGE=${CAPV_MANAGER_IMAGE:-}
+export VSPHERE_DATACENTER=${VSPHERE_DATACENTER:-}
+export VSPHERE_DATASTORE=${VSPHERE_DATASTORE:-}
+export VSPHERE_NETWORK=${VSPHERE_NETWORK:-}
+export VSPHERE_RESOURCE_POOL=${VSPHERE_RESOURCE_POOL:-}
+export VSPHERE_FOLDER=${VSPHERE_FOLDER:-}
+export VSPHERE_TEMPLATE=${VSPHERE_TEMPLATE:-}
+export VSPHERE_DISK=${VSPHERE_DISK:-}
+export VSPHERE_DISK_SIZE_GB=${VSPHERE_DISK_SIZE_GB:-20}
+export VSPHERE_NETWORK_TYPE=${VSPHERE_NETWORK_TYPE:-dhcp}
```

Comment:
Can this not be a var driven from user input? since it they choose staticIP then the rendered yaml is not sufficient?

Reply:
I might be missing something, I've been able to create VMs with `static` network type even without having the IP in the network spec. Is there something in CAPV allocating that automatically anyways?

Reply:
Hi @andrewsykim,

The image they're using will default to a fallback network if the provided config is incomplete or invalid. But that doesn't mean that we want the YAML to be/look invalid.

Unless we pivot on the network type and then change the YAML based on `static` or `dhcp`, I recommend we stick with just DHCP and let people add the rest if they want. Although doing an `if/else` shouldn't be too hard either.

The issue, then, is that for static IPs you'd need to have indexed env vars to use with each machine in the spec.

Reply:
Okay so hard code on dhcp and document how to configure with static?